chirho/indexed/internals.py

```diff
@@ -37,6 +38,21 @@ def _gather_number(
     )
 
 
+@functools.singledispatch
+def index_select_from_array_like(arr, dim: int, indices: List[int]):
```

Comment:
Don't add this here at this level, instead copy/paste the beginning part of gather and change the end that needs to instead use `np.take`

Reply:
done